backend/test_app.py

```diff
@@ -0,0 +1,271 @@
+import os
+import unittest
+import json
+from flask_sqlalchemy import SQLAlchemy
+
+from app import create_app
+from models import setup_db, Exercise, User
+
+
+class ExerciseTrackerTestCase(unittest.TestCase):
+    # This class represents the Exercise Tracker test case
+
+    def setUp(self):
+        # Define test variables and initialize app.
+        self.app = create_app()
+        self.client = self.app.test_client
+        self.database_name = "exercise-tracker_test"
+        setup_db(self.app)
+
+        self.new_user = {
+          "username": "manuelar"
+        }
+
+        self.new_exercise = {
+          "userId": "91298d0b-66c9-493b-8c1d-a79bcf7838d7",
+          "exercise_date": "Fri, 15 May 2020 00:00:00 GMT",
+          "description": "My exercise",
+          "duration": "12"
+        }
+
+        # binds the app to the current context
+        with self.app.app_context():
+            self.db = SQLAlchemy()
+            self.db.init_app(self.app)
+            # create all tables
+            self.db.create_all()
+
+    def tearDown(self):
+        # Executed after reach test
+        pass
+
+    # Tests for successful operation and for expected errors.
```

Comment:
We should re-design the tests cases. We should not make API requests in this cases because API's are seen as external objects and we cannot control what they can do. What we should do instead is that each endpoints calls the DB methods with its required parameters using DB mocks.